Web Inspector: give `WI.Spring` a better name and move it to its own file

dcrousso opened the pull request:
#### 1da03ab6270654efd0f5896e7de2e6b194572052
<pre>
Web Inspector: give `WI.Spring` a better name and move it to its own file
<a href="https://bugs.webkit.org/show_bug.cgi?id=259502">https://bugs.webkit.org/show_bug.cgi?id=259502</a>

Reviewed by Timothy Hatcher.

`WI.Spring` should really be named `WI.SpringTimingFunction` to convey it&apos;s true purpose.

It also shouldn&apos;t really be in `Source/WebInspectorUI/UserInterface/Models/Geometry.js` since it has nothing to do with geometry.

* Source/WebInspectorUI/UserInterface/Models/SpringTimingFunction.js: Added.
(WI.SpringTimingFunction):
(WI.SpringTimingFunction.fromValues):
(WI.SpringTimingFunction.fromString):
(WI.SpringTimingFunction.prototype.copy):
(WI.SpringTimingFunction.prototype.toString):
(WI.SpringTimingFunction.prototype.solve):
(WI.SpringTimingFunction.prototype.calculateDuration):
* Source/WebInspectorUI/UserInterface/Models/Geometry.js:
(WI.Spring): Deleted.
(WI.Spring.fromValues): Deleted.
(WI.Spring.fromString): Deleted.
(WI.Spring.prototype.copy): Deleted.
(WI.Spring.prototype.toString): Deleted.
(WI.Spring.prototype.solve): Deleted.
(WI.Spring.prototype.calculateDuration): Deleted.
* Source/WebInspectorUI/UserInterface/Models/Animation.js:
(WI.Animation.prototype._updateEffect):
* Source/WebInspectorUI/UserInterface/Views/AnimationContentView.js:
(WI.AnimationContentView.prototype._refreshPreview):
* Source/WebInspectorUI/UserInterface/Views/AnimationDetailsSidebarPanel.js:
(WI.AnimationDetailsSidebarPanel.prototype._refreshEffectSection):
* Source/WebInspectorUI/UserInterface/Views/SpreadsheetStyleProperty.js:
(WI.SpreadsheetStyleProperty.prototype._addTimingFunctionTokens):

* Source/WebInspectorUI/UserInterface/Views/InlineSwatch.js:
(WI.InlineSwatch.prototype.switch):
(WI.InlineSwatch.else.prototype.switch):
(WI.InlineSwatch.prototype.didDismissPopover):
(WI.InlineSwatch.prototype._fallbackValue):
(WI.InlineSwatch.prototype._swatchElementClicked):
(WI.InlineSwatch.prototype._valueEditorValueDidChange):
* Source/WebInspectorUI/UserInterface/Views/InlineSwatch.css:
(.inline-swatch:is(.bezier, .spring-timing-function, .variable)): Renamed from `.inline-swatch:is(.bezier, .spring, .variable`.
(.inline-swatch:is(.bezier, .box-shadow, .spring-timing-function, .variable)): Renamed from `.inline-swatch:is(.bezier, .box-shadow, .spring, .variable`.
(.inline-swatch:not(.read-only):is(.bezier, .box-shadow, .spring-timing-function, .variable, .alignment):hover): Renamed from `.inline-swatch:not(.read-only):is(.bezier, .box-shadow, .spring, .variable, .alignment):hover`.
(.inline-swatch:not(.read-only):is(.bezier, .box-shadow, .spring-timing-function, .variable, .alignment):active): Renamed from `.inline-swatch:not(.read-only):is(.bezier, .box-shadow, .spring, .variable, .alignment):active`.
(.inline-swatch:is(.bezier, .box-shadow, .spring-timing-function, .variable) &gt; span): Renamed from `.inline-swatch:is(.bezier, .box-shadow, .spring, .variable) &gt; span`.
Also rename `WI.InlineSwatch.Type.Spring` to `WI.InlineSwatch.Type.SpringTimingFunction`.

* Source/WebInspectorUI/UserInterface/Views/SpringTimingFunctionEditor.js: Renamed from Source/WebInspectorUI/UserInterface/Views/SpringEditor.js.
(WI.SpringTimingFunctionEditor):
(WI.SpringTimingFunctionEditor.createInputsForParameter):
(WI.SpringTimingFunctionEditor.prototype.get element):
(WI.SpringTimingFunctionEditor.prototype.get springTimingFunction):
(WI.SpringTimingFunctionEditor.prototype.set springTimingFunction):
(WI.SpringTimingFunctionEditor.prototype._handleNumberInputInput):
(WI.SpringTimingFunctionEditor.prototype._handleNumberInputKeydown):
(WI.SpringTimingFunctionEditor.prototype._handleNumberSliderInput):
(WI.SpringTimingFunctionEditor.prototype._handleNumberSliderMousedown):
(WI.SpringTimingFunctionEditor.prototype._handleNumberSliderMouseup):
(WI.SpringTimingFunctionEditor.prototype._changeSpringForInput):
(WI.SpringTimingFunctionEditor.prototype._resetPreviewAnimation):
(WI.SpringTimingFunctionEditor.prototype._updatePreviewAnimation):
* Source/WebInspectorUI/UserInterface/Views/SpringTimingFunctionEditor.css: Renamed from Source/WebInspectorUI/UserInterface/Views/SpringEditor.css.
(.spring-timing-function-editor):
(.spring-timing-function-editor &gt; .preview):
(.spring-timing-function-editor &gt; .preview &gt; div):
(.spring-timing-function-editor &gt; .timing):
(.spring-timing-function-editor &gt; .timing &gt; div):
(.spring-timing-function-editor &gt; .timing::before, .spring-timing-function-editor &gt; .timing::after):
(.spring-timing-function-editor &gt; .timing::before):
(.spring-timing-function-editor &gt; .timing::after):
(.spring-timing-function-editor &gt; :is(.preview, .timing) &gt; div):
(.spring-timing-function-editor &gt; .animate:is(.preview, .timing) &gt; div):
(.spring-timing-function-editor &gt; .number-input-container &gt; .number-input-row):
(.spring-timing-function-editor &gt; .number-input-container &gt; .number-input-row &gt; .number-input-row-title):
(.spring-timing-function-editor &gt; .number-input-container &gt; .number-input-row &gt; input):
(.spring-timing-function-editor &gt; .number-input-container &gt; .number-input-row &gt; input::-webkit-inner-spin-button):
(.spring-timing-function-editor &gt; .number-input-container &gt; .number-input-row &gt; input[type=&quot;range&quot;]):
* Source/WebInspectorUI/UserInterface/Controllers/CodeMirrorSpringTimingFunctionEditingController.js: Renamed from Source/WebInspectorUI/UserInterface/Controllers/CodeMirrorSpringEditingController.js.
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype.get initialValue):
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype.get cssClassName):
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype.popoverWillPresent):
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype.popoverDidPresent):
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype._handleSpringTimingFunctionEditorSpringTimingFunctionChanged):
Also rename `WI.SpringEditor` to `WI.SpringTimingFunctionEditor`.

* Source/WebInspectorUI/UserInterface/Models/TextMarker.js:
* Source/WebInspectorUI/UserInterface/Views/CodeMirrorTextMarkers.js:
(createCodeMirrorSpringTimingFunctionTextMarkers): Renamed from `createCodeMirrorSpringTextMarkers`.
* Source/WebInspectorUI/UserInterface/Views/SourceCodeTextEditor.js:
(WI.SourceCodeTextEditor.prototype._updateEditableMarkers):
* Source/WebInspectorUI/UserInterface/Views/TextEditor.js:
(WI.TextEditor.prototype.createSpringTimingFunctionMarkers): Renamed from `createSpringMarkers`.
(WI.TextEditor.prototype.editingControllerForMarker):
Also rename `WI.TextMarker.Type.Spring` to `WI.TextMarker.Type.SpringTimingFunction`.

* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Test.html:
* Source/WebInspectorUI/.eslintrc:

* LayoutTests/inspector/animation/resources/lifecycle-utilities.js:
* LayoutTests/inspector/unit-tests/geometry.html:
* LayoutTests/inspector/unit-tests/geometry-expected.txt:
* LayoutTests/inspector/unit-tests/spring-timing-function.html: Added.
* LayoutTests/inspector/unit-tests/spring-timing-function-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/266338@main">https://commits.webkit.org/266338@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/598326ae1f1f335ea4c6e53fd73159ac30b02e58

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13572 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13886 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14217 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15309 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12901 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16394 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13964 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15586 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13738 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14386 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11486 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16013 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11674 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12241 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19293 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12749 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12409 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/15630 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12930 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/10815 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12197 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/12106 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3304 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16527 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12773 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rcaliman-apple commented on Source/WebInspectorUI/UserInterface/Views/SpringTimingFunctionEditor.js:
```diff
@@ -23,27 +23,22 @@
+        this._previewContainer.addEventListener("mousedown", this._resetPreviewAnimation.bind(this));
 
         this._previewElement = this._previewContainer.createChild("div");
-        this._previewElement.addEventListener("transitionend", boundResetPreviewAnimation);
```
Do we no longer need to reset the preview on "transitionend" or was this not necessary even before this patch?

dcrousso replied:
oh hmm no idea how that happened.  ill revert

rcaliman-apple commented on Source/WebInspectorUI/UserInterface/Models/SpringTimingFunction.js:
```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2013 Apple Inc. All rights reserved.
```
Should we update the year?

dcrousso replied:
i would say no since im not really modifying the actual code, so IMO it should match the same copyright it had before

rcaliman-apple reviewed (approved):
Looks good to me 👍. Just a question inline. 
Also, should we update the Copyright year in the new and renamed files?

Pinging @patrickangle for review / approval.

dcrousso committed:
Web Inspector: give `WI.Spring` a better name and move it to its own file
https://bugs.webkit.org/show_bug.cgi?id=259502

Reviewed by Timothy Hatcher.

`WI.Spring` should really be named `WI.SpringTimingFunction` to convey it's true purpose.

It also shouldn't really be in `Source/WebInspectorUI/UserInterface/Models/Geometry.js` since it has nothing to do with geometry.

* Source/WebInspectorUI/UserInterface/Models/SpringTimingFunction.js: Added.
(WI.SpringTimingFunction):
(WI.SpringTimingFunction.fromValues):
(WI.SpringTimingFunction.fromString):
(WI.SpringTimingFunction.prototype.copy):
(WI.SpringTimingFunction.prototype.toString):
(WI.SpringTimingFunction.prototype.solve):
(WI.SpringTimingFunction.prototype.calculateDuration):
* Source/WebInspectorUI/UserInterface/Models/Geometry.js:
(WI.Spring): Deleted.
(WI.Spring.fromValues): Deleted.
(WI.Spring.fromString): Deleted.
(WI.Spring.prototype.copy): Deleted.
(WI.Spring.prototype.toString): Deleted.
(WI.Spring.prototype.solve): Deleted.
(WI.Spring.prototype.calculateDuration): Deleted.
* Source/WebInspectorUI/UserInterface/Models/Animation.js:
(WI.Animation.prototype._updateEffect):
* Source/WebInspectorUI/UserInterface/Views/AnimationContentView.js:
(WI.AnimationContentView.prototype._refreshPreview):
* Source/WebInspectorUI/UserInterface/Views/AnimationDetailsSidebarPanel.js:
(WI.AnimationDetailsSidebarPanel.prototype._refreshEffectSection):
* Source/WebInspectorUI/UserInterface/Views/SpreadsheetStyleProperty.js:
(WI.SpreadsheetStyleProperty.prototype._addTimingFunctionTokens):

* Source/WebInspectorUI/UserInterface/Views/InlineSwatch.js:
(WI.InlineSwatch.prototype.switch):
(WI.InlineSwatch.else.prototype.switch):
(WI.InlineSwatch.prototype.didDismissPopover):
(WI.InlineSwatch.prototype._fallbackValue):
(WI.InlineSwatch.prototype._swatchElementClicked):
(WI.InlineSwatch.prototype._valueEditorValueDidChange):
* Source/WebInspectorUI/UserInterface/Views/InlineSwatch.css:
(.inline-swatch:is(.bezier, .spring-timing-function, .variable)): Renamed from `.inline-swatch:is(.bezier, .spring, .variable`.
(.inline-swatch:is(.bezier, .box-shadow, .spring-timing-function, .variable)): Renamed from `.inline-swatch:is(.bezier, .box-shadow, .spring, .variable`.
(.inline-swatch:not(.read-only):is(.bezier, .box-shadow, .spring-timing-function, .variable, .alignment):hover): Renamed from `.inline-swatch:not(.read-only):is(.bezier, .box-shadow, .spring, .variable, .alignment):hover`.
(.inline-swatch:not(.read-only):is(.bezier, .box-shadow, .spring-timing-function, .variable, .alignment):active): Renamed from `.inline-swatch:not(.read-only):is(.bezier, .box-shadow, .spring, .variable, .alignment):active`.
(.inline-swatch:is(.bezier, .box-shadow, .spring-timing-function, .variable) > span): Renamed from `.inline-swatch:is(.bezier, .box-shadow, .spring, .variable) > span`.
Also rename `WI.InlineSwatch.Type.Spring` to `WI.InlineSwatch.Type.SpringTimingFunction`.

* Source/WebInspectorUI/UserInterface/Views/SpringTimingFunctionEditor.js: Renamed from Source/WebInspectorUI/UserInterface/Views/SpringEditor.js.
(WI.SpringTimingFunctionEditor):
(WI.SpringTimingFunctionEditor.createInputsForParameter):
(WI.SpringTimingFunctionEditor.prototype.get element):
(WI.SpringTimingFunctionEditor.prototype.get springTimingFunction):
(WI.SpringTimingFunctionEditor.prototype.set springTimingFunction):
(WI.SpringTimingFunctionEditor.prototype._handleNumberInputInput):
(WI.SpringTimingFunctionEditor.prototype._handleNumberInputKeydown):
(WI.SpringTimingFunctionEditor.prototype._handleNumberSliderInput):
(WI.SpringTimingFunctionEditor.prototype._handleNumberSliderMousedown):
(WI.SpringTimingFunctionEditor.prototype._handleNumberSliderMouseup):
(WI.SpringTimingFunctionEditor.prototype._changeSpringForInput):
(WI.SpringTimingFunctionEditor.prototype._resetPreviewAnimation):
(WI.SpringTimingFunctionEditor.prototype._updatePreviewAnimation):
* Source/WebInspectorUI/UserInterface/Views/SpringTimingFunctionEditor.css: Renamed from Source/WebInspectorUI/UserInterface/Views/SpringEditor.css.
(.spring-timing-function-editor):
(.spring-timing-function-editor > .preview):
(.spring-timing-function-editor > .preview > div):
(.spring-timing-function-editor > .timing):
(.spring-timing-function-editor > .timing > div):
(.spring-timing-function-editor > .timing::before, .spring-timing-function-editor > .timing::after):
(.spring-timing-function-editor > .timing::before):
(.spring-timing-function-editor > .timing::after):
(.spring-timing-function-editor > :is(.preview, .timing) > div):
(.spring-timing-function-editor > .animate:is(.preview, .timing) > div):
(.spring-timing-function-editor > .number-input-container > .number-input-row):
(.spring-timing-function-editor > .number-input-container > .number-input-row > .number-input-row-title):
(.spring-timing-function-editor > .number-input-container > .number-input-row > input):
(.spring-timing-function-editor > .number-input-container > .number-input-row > input::-webkit-inner-spin-button):
(.spring-timing-function-editor > .number-input-container > .number-input-row > input[type="range"]):
* Source/WebInspectorUI/UserInterface/Controllers/CodeMirrorSpringTimingFunctionEditingController.js: Renamed from Source/WebInspectorUI/UserInterface/Controllers/CodeMirrorSpringEditingController.js.
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype.get initialValue):
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype.get cssClassName):
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype.popoverWillPresent):
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype.popoverDidPresent):
(WI.CodeMirrorSpringTimingFunctionEditingController.prototype._handleSpringTimingFunctionEditorSpringTimingFunctionChanged):
Also rename `WI.SpringEditor` to `WI.SpringTimingFunctionEditor`.

* Source/WebInspectorUI/UserInterface/Models/TextMarker.js:
* Source/WebInspectorUI/UserInterface/Views/CodeMirrorTextMarkers.js:
(createCodeMirrorSpringTimingFunctionTextMarkers): Renamed from `createCodeMirrorSpringTextMarkers`.
* Source/WebInspectorUI/UserInterface/Views/SourceCodeTextEditor.js:
(WI.SourceCodeTextEditor.prototype._updateEditableMarkers):
* Source/WebInspectorUI/UserInterface/Views/TextEditor.js:
(WI.TextEditor.prototype.createSpringTimingFunctionMarkers): Renamed from `createSpringMarkers`.
(WI.TextEditor.prototype.editingControllerForMarker):
Also rename `WI.TextMarker.Type.Spring` to `WI.TextMarker.Type.SpringTimingFunction`.

* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Test.html:
* Source/WebInspectorUI/.eslintrc:

* LayoutTests/inspector/animation/resources/lifecycle-utilities.js:
* LayoutTests/inspector/unit-tests/geometry.html:
* LayoutTests/inspector/unit-tests/geometry-expected.txt:
* LayoutTests/inspector/unit-tests/spring-timing-function.html: Added.
* LayoutTests/inspector/unit-tests/spring-timing-function-expected.txt: Added.

Canonical link: https://commits.webkit.org/266338@main
